src/LibVLCSharp.Eto/VideoView.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Eto.Drawing;
+using Eto.Forms;
+using LibVLCSharp.Shared;
+
+namespace LibVLCSharp.Eto
+{
+    /// <summary>
+    /// Eto.Forms VideoView control with a LibVLCSharp MediaPlayer
+    /// </summary>
+    public class VideoView : Panel, IVideoView, IDisposable
+    {
+        /// <summary>
+        /// The VideoView constructor.
+        /// </summary>
+        public VideoView()
+        {
+            BackgroundColor = Colors.Black;
+        }
+
+        MediaPlayer? _mp;
+
+        /// <summary>
+        /// The MediaPlayer attached to this view (or null)
+        /// </summary>
+        public MediaPlayer? MediaPlayer
+        {
+            get => _mp;
+            set
+            {
+                if (ReferenceEquals(_mp, value))
+                {
+                    return;
+                }
+
+                MpDetach();
+                _mp = value;
+                MpAttach();
+            }
+        }
+
+        void MpDetach()
+        {
+            if (_mp == null || _mp.NativeReference == IntPtr.Zero)
+                return;
+
+            _mp.Hwnd = IntPtr.Zero;
+        }
+
+        void MpAttach()
+        {
+            if (_mp == null || _mp.NativeReference == IntPtr.Zero)
+                return;
+
+            if (NativeHandle == IntPtr.Zero)
+                throw new InvalidOperationException("VideoView Control must be shown before assigning a MediaPlayer");
```

Comment:
think that might be a bit strict, we don't currently throw in other integrations in this case. We might revisit that later, but for now this should be aligned with how other platforms behave.